File: web/app/types/actor.ts
// Lightweight actor for list views
export interface ActorListItem {
    id: number;
    uuid: string;
    name: string;
    aliases: string[];
    image_url: string;
    gender: string;
    scene_count: number;
}

// Full actor details for detail views
export interface Actor {
    id: number;
    uuid: string;
    created_at: string;
    updated_at: string;
    name: string;
    aliases?: string[];
    image_url?: string;
    gender?: string;
    birthday?: string;
    date_of_death?: string;
    astrology?: string;
    birthplace?: string;
    ethnicity?: string;
    nationality?: string;
    career_start_year?: number;
    career_end_year?: number;
    height_cm?: number;
    weight_kg?: number;
    measurements?: string;
    cupsize?: string;
    hair_color?: string;
    eye_color?: string;
    tattoos?: string;
    piercings?: string;
    fake_boobs: boolean;
    same_sex_only: boolean;
    scene_count?: number;
}

export interface ActorListResponse {
    data: ActorListItem[];
    total: number;
    page: number;
    limit: number;
}

export interface CreateActorInput {
    name: string;
    aliases?: string[];
    image_url?: string;
    gender?: string;
    birthday?: string;
    date_of_death?: string;
    astrology?: string;
    birthplace?: string;
    ethnicity?: string;
    nationality?: string;
    career_start_year?: number;
    career_end_year?: number;
    height_cm?: number;
    weight_kg?: number;
    measurements?: string;
    cupsize?: string;
    hair_color?: string;
    eye_color?: string;
    tattoos?: string;
    piercings?: string;
    fake_boobs?: boolean;
    same_sex_only?: boolean;
}

export interface UpdateActorInput {
    name?: string;
    aliases?: string[];
    image_url?: string;
    gender?: string;
    birthday?: string;
    date_of_death?: string;
    astrology?: string;
    birthplace?: string;
    ethnicity?: string;
    nationality?: string;
    career_start_year?: number;
    career_end_year?: number;
    height_cm?: number;
    weight_kg?: number;
    measurements?: string;
    cupsize?: string;
    hair_color?: string;
    eye_color?: string;
    tattoos?: string;
    piercings?: string;
    fake_boobs?: boolean;
    same_sex_only?: boolean;
}

File: internal/api/v1/request/actor.go
package request

type CreateActorRequest struct {
	Name            string   `json:"name" binding:"required"`
	Aliases         []string `json:"aliases"`
	ImageURL        string   `json:"image_url"`
	Gender          string   `json:"gender"`
	Birthday        *string `json:"birthday"`
	DateOfDeath     *string `json:"date_of_death"`
	Astrology       string  `json:"astrology"`
	Birthplace      string  `json:"birthplace"`
	Ethnicity       string  `json:"ethnicity"`
	Nationality     string  `json:"nationality"`
	CareerStartYear *int    `json:"career_start_year"`
	CareerEndYear   *int    `json:"career_end_year"`
	HeightCm        *int    `json:"height_cm"`
	WeightKg        *int    `json:"weight_kg"`
	Measurements    string  `json:"measurements"`
	Cupsize         string  `json:"cupsize"`
	HairColor       string  `json:"hair_color"`
	EyeColor        string  `json:"eye_color"`
	Tattoos         string  `json:"tattoos"`
	Piercings       string  `json:"piercings"`
	FakeBoobs       bool    `json:"fake_boobs"`
	SameSexOnly     bool    `json:"same_sex_only"`
}

type UpdateActorRequest struct {
	Name            *string   `json:"name"`
	Aliases         *[]string `json:"aliases"`
	ImageURL        *string   `json:"image_url"`
	Gender          *string `json:"gender"`
	Birthday        *string `json:"birthday"`
	DateOfDeath     *string `json:"date_of_death"`
	Astrology       *string `json:"astrology"`
	Birthplace      *string `json:"birthplace"`
	Ethnicity       *string `json:"ethnicity"`
	Nationality     *string `json:"nationality"`
	CareerStartYear *int    `json:"career_start_year"`
	CareerEndYear   *int    `json:"career_end_year"`
	HeightCm        *int    `json:"height_cm"`
	WeightKg        *int    `json:"weight_kg"`
	Measurements    *string `json:"measurements"`
	Cupsize         *string `json:"cupsize"`
	HairColor       *string `json:"hair_color"`
	EyeColor        *string `json:"eye_color"`
	Tattoos         *string `json:"tattoos"`
	Piercings       *string `json:"piercings"`
	FakeBoobs       *bool   `json:"fake_boobs"`
	SameSexOnly     *bool   `json:"same_sex_only"`
}

type SetSceneActorsRequest struct {
	ActorIDs []uint `json:"actor_ids"`
}
